lib/improver/tests/spotdata/spotdata/test_ancillaries.py

```diff
@@ -0,0 +1,175 @@
+
+        cube = Cube(data,
+                    long_name="template",
+                    dim_coords_and_dims=[(latitude, 0), (longitude, 1)],
+                    units="1")
+
+        orography = cube.copy()
+        orography.data[0:10, 0:10] = 2.
+        orography.rename('surface_altitude')
+        land = cube.copy()
+        land.rename('land_binary_mask')
+        land.data = land.data + 1
+
+        self.orography = orography
+        self.land = land
+        self.directory = mkdtemp()
+        self.orography_path = self.directory + '/orography.nc'
+        self.land_path = self.directory + '/land_mask.nc'
+        iris.save(orography, self.orography_path)
+        iris.save(land, self.land_path)
+
+    def tearDown(self):
+        """Remove temporary directories created for testing."""
+        Call(['rm', '-rf', self.directory])
+
+    def test_return_type(self):
+        """
+        Test return type is a dictionary containing cubes.
+
+        """
+        diagnostics = {}
+        result = Plugin(diagnostics, self.directory)
+        self.assertIsInstance(result, dict)
+
+    def test_read_orography(self):
+        """
+        Test reading an orography netcdf file.
+
+        """
+        diagnostics = {}
+        result = Plugin(diagnostics, self.directory)
+        self.assertIn('orography', result.keys())
+        self.assertIsInstance(result['orography'], Cube)
+        self.assertArrayEqual(result['orography'].data, self.orography.data)
+
+    def test_read_land_mask(self):
+        """
+        Test reading a landmask netcdf file if a diagnostic makes use of a land
+        constraint condition.
+
+        """
+        diagnostics = {
+            "wind_speed": {
+                "diagnostic_name": "wind_speed",
+                "extrema": False,
+                "filepath": "horizontal_wind_speed_and_direction_at_10m",
+                "interpolation_method": "use_nearest",
+                "neighbour_finding": {
+                    "land_constraint": True,
+                    "method": "fast_nearest_neighbour",
+                    "vertical_bias": None
+                    }
+                }
+            }
+
+        result = Plugin(diagnostics, self.directory)
+        self.assertIn('land_mask', result.keys())
+        self.assertIsInstance(result['land_mask'], Cube)
+        self.assertArrayEqual(result['land_mask'].data, self.land.data)
+
+    def test_missing_orography(self):
+        """
+        Test attempt to read orography with missing orography netcdf file.
+
+        """
+        Call(['rm', self.orography_path])
+        diagnostics = {}
+        msg = 'Orography file not found.'
+        with self.assertRaisesRegexp(IOError, msg):
+            Plugin(diagnostics, self.directory)
+
+    def test_missing_land_mask(self):
+        """
+        Test attempt to read land_mask with missing land_mask netcdf file.
+
+        """
+        Call(['rm', self.land_path])
+
+        diagnostics = {
```

Comment:
You could define this as `self.diagnostics` in setUp to avoid duplication with `test_read_landmask`.